piopio/piopio_be/views.py

```diff
@@ -178,13 +178,19 @@ def related(self, request, userpk):
             print(_user)
             related.append(_user['id'])
 
-        #user = request.user
         related.append(userpk)
         posts = models.Post.objects.filter(
             user_id__in=related).order_by('-created_at')
         posts = add_likes_and_retweets(posts, userpk)
 
-        page = self.paginate_queryset(posts)
+        """
+        Filter the results with user's report list
+        """
+        user = self.queryset.get(pk=request.user.pk)
+        reported = user.reported.all().values_list('id')
+        filtered = posts.all().exclude(id__in=reported)
```

Comment:
The posts should also be filtered out if the post has more than 10 reports in total.

Reply:
I'll keep this part, the simple reason is that by now we only have to implement a "fake" report function (Also I'd like to point out that adding a counter to the report function will make the demo way more difficult cuz we need to have log in 10 different account to report the same post).